src/Service.Host/client/src/components/StockPools.js

```diff
@@ -0,0 +1,306 @@
+                            search={searchStorageLocations}
+                            searchResults={storageLocationsSearchResults}
+                            priorityFunction="closestMatchesFirst"
+                            onResultSelect={handleSearchResultSelect}
+                            clearSearch={clearStorageLocation}
+                        />
+                    </DialogContent>
+                    <DialogActions>
+                        <Button onClick={handleClose}>Close</Button>
+                    </DialogActions>
+                </Dialog>
+            </div>
+        );
+    };
+
+    const stockPoolColumns = [
+        {
+            field: 'stockPoolCode',
+            headerName: 'Code',
+            editable: params => params.row?.creating === true,
+            width: 100
+        },
+        { field: 'stockPoolDescription', editable: true, headerName: 'Description', width: 225 },
+        {
+            field: 'dateInvalid',
+            headerName: 'Date Invalid',
+            editable: true,
+            width: 175,
+            valueFormatter: value => value && moment(value).format('DD-MMM-YYYY')
+        },
+        { field: 'stockCategory', editable: true, headerName: 'Stock Category', width: 130 },
+        {
+            field: 'accountingCompanyCode',
+            editable: true,
+            headerName: 'Accounting Company',
+            width: 175,
+            defaultFormat: true,
+            valueOptions: accountingCompany?.map(ac => ac.name),
+            type: 'singleSelect'
+        },
+        {
+            field: 'defaultLocationName',
+            headerName: 'Default Location',
+            width: 150,
+            editable: true,
+            type: 'search',
+            searchResults: storageLocationsSearchResults,
+            searchLoading: storageLocationsSearchLoading,
+            searchUpdateFieldNames: [
+                { fieldName: 'defaultLocation', searchResultFieldName: 'locationId' },
+                { fieldName: 'defaultLocationName', searchResultFieldName: 'locationCode' }
+            ],
+            clearSearch: clearStorageLocation,
+            renderCell: searchRenderCell
+        },
+        { field: 'availableToMrp', editable: true, headerName: 'Available to Mrp', width: 175 },
+        { field: 'sequence', editable: true, headerName: 'Sequence', width: 100 }
+    ];
+
+    const sortedStockPools = stockPools?.slice().sort((a, b) => {
+        const fa = a.sequence;
+        const fb = b.sequence;
+
+        if (fa === null && fb !== null) {
+            return 1;
+        }
+        if (fa !== null && fb === null) {
+            return -1;
+        }
+        if (fa === null && fb === null) {
+            return 0;
+        }
+        return fa - fb;
+    });
+
+    return (
+        <Page homeUrl={config.appRoot} showAuthUi={false}>
+            {stockPoolColumns?.filter(c => c.type === 'search').map(c => renderSearchDialog(c))}
```

Comment:
Don't need this filter and map either - It looks like a lot of this code is copied from an example which needed to the flexibility to handle mutliple custom column types in one datagrid - you only have one Search one here so no need for this additional complexity again